### ISSUE TITLE
Adding PINN code

### ISSUE THREAD

**Issue by asztalosors** [plasma-sds/bes-inference#6]
PINN features can be added to any NN, by the inclusion of a physics-based loss function in addition to other loss functions.  For BES inference-related issues, the loss function will be generated by assembling the rate coefficient matrix, which requires prior knowledge of beam species, energy Zeff and a temperature profile. 

Rate loading and processing will be done with the use of renate-od related utility functions, dedicated for BES synthetic diagnostics.

**Comment by asztalosors** [comment]
The introduced PINN is currently not being developed, and PINN features would be explored not by introducing tailored loss functions but by combining modeling-based training sets and experimental datasets. The existing code would be placed in a dedicated branch #18 as part of the ongoing code restructuring in Issue #16
